contracts/mock/ConversionRatesEnhancedOpen.sol

```diff
@@ -0,0 +1,93 @@
+pragma solidity 0.4.18;
+
+import "../reserves/fprConversionRate/ConversionRateEnhancedSteps.sol";
+
+
+contract ConversionRateEnhancedOpen is ConversionRateEnhancedSteps {
+
+    function ConversionRateEnhancedOpen(address _admin) public 
+        ConversionRateEnhancedSteps(_admin)
+        { } // solhint-disable-line no-empty-blocks
+
+    function getRateOpenData(
+        ERC20 token, 
+        uint currentBlockNumber, 
+        bool buy, 
+        uint qty 
+    ) public view returns(
+        uint rateWithSteps,
+        uint rateWithoutSteps,
+        int extraBpsYdata
+    ) {
+        // get rate update block
+        bytes32 compactData = tokenRatesCompactData[tokenData[token].compactDataArrayIndex];
+
+        uint updateRateBlock = getLast4Bytes(compactData);
+        if (currentBlockNumber >= updateRateBlock + validRateDurationInBlocks) return (0, 0, 0); // rate is expired
+        // check imbalance
+        int totalImbalance;
+        int blockImbalance;
+        (totalImbalance, blockImbalance) = getImbalance(token, updateRateBlock, currentBlockNumber);
+
+        return getRateDataFakeImbalance(token, buy, qty, totalImbalance);
+    }
+
+    function getRateDataFakeImbalance(
```

Comment:
We don't have check for `totalImbalance` (or `fakeImbalance`) and `blockImbalance`, is it our intention?